ravenframework/TSA/ARMA.py

```diff
@@ -132,15 +176,18 @@ def setDefaults(self, settings):
       settings['engine'] = randomUtils.newRNG()
     if 'reduce_memory' not in settings:
       settings['reduce_memory'] = False
+    if 'auto_select' not in settings:
+      settings['auto_select'] = False
     return settings
 
-  def fit(self, signal, pivot, targets, settings):
+  def fit(self, signal, pivot, targets, settings, trainedParams=None):
     """
       Determines the charactistics of the signal based on this algorithm.
       @ In, signal, np.ndarray, time series with dims [time, target]
       @ In, pivot, np.1darray, time-like parameter values
       @ In, targets, list(str), names of targets in same order as signal
       @ In, settings, dict, settings for this ROM
+      @ In, trainedParams, dict, running dict of trained algorithm params
       @ Out, params, dict, characteristic parameters
     """
     # lazy import statsmodels
```

Comment:
If the fit function was called several times should we import each time? I am a little against having the import here. I know that this is not added by you, but do you think we'd better move it?

Reply:
Python will only import it the first time, and after that the import is cached. imports inside of function means that the import only happens when the function is called, which can speed up RAVEN since otherwise the import needs to happen at the start of RAVEN even if statsmodels is never used.

Reply:
Thanks Josh. I am saying this to make you aware of it, as I have been directed not to do that before, but if you are aware of it, I am totally OK.